diff --git a/dist/index.mjs b/dist/index.mjs
index 8a688331b88b4af738ee4ca8062b5f24124d3d81..c0095c3d33711d9b9998988da4d4927b3a5cc11d 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2028,7 +2028,9 @@ var OpenRouterChatLanguageModel = class {
                   const lastDetail = accumulatedReasoningDetails[accumulatedReasoningDetails.length - 1];
                   if ((lastDetail == null ? void 0 : lastDetail.type) === "reasoning.text" /* Text */) {
                     lastDetail.text = (lastDetail.text || "") + (detail.text || "");
-                    lastDetail.signature = lastDetail.signature || detail.signature;
+                    if (lastDetail.signature || detail.signature) {
+                      lastDetail.signature = lastDetail.signature || detail.signature;
+                    }
                     lastDetail.format = lastDetail.format || detail.format;
                   } else {
                     accumulatedReasoningDetails.push(__spreadValues({}, detail));
